### PR TITLE
Removed composer.lock

### PR DESCRIPTION
**Pull request by joaorobertopb**
- Added composer.lock to .gitignore file
- Organizing .gitignore file

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,9 +1,8 @@
-/vendor/
-.idea/*
 
+.idea/*
 .idea/codeStyleSettings.xml
-
 .twgit_features_subject
-
+composer.lock
```
Comment (nunomaduro) [inline .gitignore]: The composer.lock should always be committed.

Comment (thecrypticace) [reply .gitignore]: No, not for packages. When used in a package it only serves to speed up CI builds. The projects that require this package will ignore the lock file entirely. Only a top-level lock file is ever used by Composer. This also means that CI won't run against the latest dependencies as would happen when used in actual projects / applications.

Comment (joaorobertopb) [reply .gitignore]: It is not common committed composer.lock in packages. 

@thecrypticace great explanation!

Comment (nunomaduro) [reply .gitignore]: @joaorobertopb @thecrypticace You guys are both right. Thanks for the explanation!

**Comment by nunomaduro** [comment]
Thanks for the contribution! 👍

**Pull request merged**